app/src/App.tsx

```diff
@@ -84,10 +80,9 @@ export const App = () => {
   return (
     <React.Fragment>
       {
-        isInitializing
+        isInitializing 
           ? <div>Loading...</div>
-          : isAuthenticated
-            // ? hasAccount
+          : isAuthenticated && hasAccount
```

Comment:
Works great. We can possibly improve the code quality in the future by refactoring the state + fetch + history.push interaction so we don't need this structure